TextTrack should be constructed with a ScriptExecutionContext rather than a Document

jyavenard opened the pull request:
#### 21b659583b5f8410887b7fada7bf8775ced3f3db
<pre>
TextTrack should be constructed with a ScriptExecutionContext rather than a Document
<a href="https://bugs.webkit.org/show_bug.cgi?id=269181">https://bugs.webkit.org/show_bug.cgi?id=269181</a>
<a href="https://rdar.apple.com/122776217">rdar://122776217</a>

Reviewed by Eric Carlson and Ryosuke Niwa.

In order to have MSE in a Worker, we&apos;ll need to have {Text/Audio/Video} to be available in DedicatedWorker.

This change just the TextTracks constructors to take a ScriptExecutionContext ; it still requires it
to be a Document for now.

* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment):
(WebCore::SourceBuffer::document const): Deleted.
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/html/HTMLTrackElement.h:
* Source/WebCore/html/track/InbandDataTextTrack.cpp:
(WebCore::InbandDataTextTrack::InbandDataTextTrack):
(WebCore::InbandDataTextTrack::create):
(WebCore::InbandDataTextTrack::addDataCue):
* Source/WebCore/html/track/InbandDataTextTrack.h:
* Source/WebCore/html/track/InbandGenericTextTrack.cpp:
(WebCore::InbandGenericTextTrack::InbandGenericTextTrack):
(WebCore::InbandGenericTextTrack::create):
(WebCore::InbandGenericTextTrack::addGenericCue):
(WebCore::InbandGenericTextTrack::parser):
(WebCore::InbandGenericTextTrack::newCuesParsed):
* Source/WebCore/html/track/InbandGenericTextTrack.h:
* Source/WebCore/html/track/InbandTextTrack.cpp:
(WebCore::InbandTextTrack::create):
(WebCore::InbandTextTrack::InbandTextTrack):
* Source/WebCore/html/track/InbandTextTrack.h:
* Source/WebCore/html/track/InbandWebVTTTextTrack.cpp:
(WebCore::InbandWebVTTTextTrack::InbandWebVTTTextTrack):
(WebCore::InbandWebVTTTextTrack::create):
(WebCore::InbandWebVTTTextTrack::parser):
(WebCore::InbandWebVTTTextTrack::newCuesParsed):
* Source/WebCore/html/track/InbandWebVTTTextTrack.h:
* Source/WebCore/html/track/LoadableTextTrack.cpp:
(WebCore::LoadableTextTrack::LoadableTextTrack):
* Source/WebCore/html/track/TextTrack.cpp:
(WebCore::TextTrack::create):
(WebCore::TextTrack::document const): Deleted.
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::prepareToSpeak):

Canonical link: <a href="https://commits.webkit.org/274455@main">https://commits.webkit.org/274455@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/389ca12f71fa4cedbee37e7652c7c0e961e24fed

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39166 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18129 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41519 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41700 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34883 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41472 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/20995 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15474 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32780 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39740 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15265 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33950 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13260 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/13227 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34888 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42977 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35561 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35217 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39044 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13978 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11533 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/37275 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15584 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8760 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/15247 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/15070 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/html/track/InbandDataTextTrack.cpp:
```diff
@@ -54,17 +54,24 @@ InbandDataTextTrack::~InbandDataTextTrack() = default;
 void InbandDataTextTrack::addDataCue(const MediaTime& start, const MediaTime& end, const void* data, unsigned length)
 {
-    addCue(DataCue::create(document(), start, end, data, length));
+    // TODO: handle datacue creation on worker.
```
Please use FIXME instead.

rniwa commented on Source/WebCore/html/track/InbandDataTextTrack.cpp:
```diff
@@ -54,17 +54,24 @@ InbandDataTextTrack::~InbandDataTextTrack() = default;
 
 void InbandDataTextTrack::addDataCue(const MediaTime& start, const MediaTime& end, Ref<SerializedPlatformDataCue>&& platformValue, const String& type)
 {
+    // TODO: handle datacue creation on worker.
```
Ditto.

jyavenard committed:
TextTrack should be constructed with a ScriptExecutionContext rather than a Document
https://bugs.webkit.org/show_bug.cgi?id=269181
rdar://122776217

Reviewed by Eric Carlson and Ryosuke Niwa.

In order to have MSE in a Worker, we'll need to have {Text/Audio/Video} to be available in DedicatedWorker.

This change just the TextTracks constructors to take a ScriptExecutionContext ; it still requires it
to be a Document for now.

* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment):
(WebCore::SourceBuffer::document const): Deleted.
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/html/HTMLTrackElement.h:
* Source/WebCore/html/track/InbandDataTextTrack.cpp:
(WebCore::InbandDataTextTrack::InbandDataTextTrack):
(WebCore::InbandDataTextTrack::create):
(WebCore::InbandDataTextTrack::addDataCue):
* Source/WebCore/html/track/InbandDataTextTrack.h:
* Source/WebCore/html/track/InbandGenericTextTrack.cpp:
(WebCore::InbandGenericTextTrack::InbandGenericTextTrack):
(WebCore::InbandGenericTextTrack::create):
(WebCore::InbandGenericTextTrack::addGenericCue):
(WebCore::InbandGenericTextTrack::parser):
(WebCore::InbandGenericTextTrack::newCuesParsed):
* Source/WebCore/html/track/InbandGenericTextTrack.h:
* Source/WebCore/html/track/InbandTextTrack.cpp:
(WebCore::InbandTextTrack::create):
(WebCore::InbandTextTrack::InbandTextTrack):
* Source/WebCore/html/track/InbandTextTrack.h:
* Source/WebCore/html/track/InbandWebVTTTextTrack.cpp:
(WebCore::InbandWebVTTTextTrack::InbandWebVTTTextTrack):
(WebCore::InbandWebVTTTextTrack::create):
(WebCore::InbandWebVTTTextTrack::parser):
(WebCore::InbandWebVTTTextTrack::newCuesParsed):
* Source/WebCore/html/track/InbandWebVTTTextTrack.h:
* Source/WebCore/html/track/LoadableTextTrack.cpp:
(WebCore::LoadableTextTrack::LoadableTextTrack):
* Source/WebCore/html/track/TextTrack.cpp:
(WebCore::TextTrack::create):
(WebCore::TextTrack::document const): Deleted.
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::prepareToSpeak):

Canonical link: https://commits.webkit.org/274455@main
